src/writer/api/clients.py

```
@@ -0,0 +1,158 @@
+
+    def set_value(self, key: str, value: Any, user_id: Optional[str] = None):
+        """
+        Set a key-value pair in the agent data.
+        :param key: The key to set.
+        :param value: The value to set for the key.
+        :param user_id: Optional user ID for the request.
+        :return: The response from the API.
+        """
+        response = self._set(key, value, user_id)
+        if response is None:
+            logging.error(f"Failed to set value for key: {key}")
```

Pylint (3.3.7)

[warning] Use lazy % formatting in logging functions

(W1203)